### PR TITLE
added get_settings method to both sync and async clients + tests

### PR DESCRIPTION
**Pull request by MarvaZy**

### CODE REVIEW

File: heksher/clients/async_client.py [hunk 1]
```diff
@@ -192,3 +194,13 @@ async def ping(self) -> None:
...
         response = await self._http_client.get('/api/health')
         response.raise_for_status()
+
+    async def get_settings(self) -> List[SettingData]:
+        """
+        List all the settings in the service
+        """
+        request_data = {'include_additional_data': True}
+        response = await self._http_client.get('/api/v1/settings', params=json.dumps(request_data))
+        settings = GetSettingsOutputWithData.parse_obj(response.json())
```
Comment (aviramha) [inline heksher/clients/async_client.py:204]: orjson

File: tests/unittest/test_async_client.py [hunk 2]
```diff
@@ -259,3 +261,21 @@ async def test_health_unreachable():
...
+            'configurable_features': ['a', 'b'],
+            'type': 'test_type',
+            'default_value': None,
+            'metadata': {}
+        }
+    ]}
+    monkeypatch.setattr(fake_heksher_service, 'settings_response', orjson.dumps(settings))
+    async with AsyncHeksherClient(fake_heksher_service.url, 10000000, ['a', 'b', 'c']) as client:
+        response = await client.get_settings()
+        assert response == [SettingData(**setting) for setting in settings['settings']]
```
Comment (bentheiii) [inline tests/unittest/test_async_client.py:281]: actually create the expected settings and test against them

File: pyproject.toml [hunk 3]
```diff
@@ -16,6 +16,7 @@ orjson = "^3.0.0"
 httpx = "^0.16.0"
 mock = {version="^4.0.0", markers = "python_version < '3.8'"}
 ordered-set = "^4.0.0"
+pydantic = "^1.8.1"
```
Comment (bentheiii) [inline pyproject.toml:19]: 1.0.0

File: heksher/clients/util.py [hunk 4]
```diff
@@ -1,10 +1,16 @@
 from logging import getLogger
+from dataclasses import dataclass
+
+import orjson
 from operator import itemgetter
-from typing import Iterable, Tuple, Sequence, TypeVar
+from typing import Iterable, Tuple, Sequence, TypeVar, List, Any, Dict
```
Comment (bentheiii) [inline heksher/clients/util.py:6]: importssss

File: tests/unittest/test_async_client.py [hunk 5]
```diff
@@ -278,10 +278,9 @@ async def test_get_settings(fake_heksher_service, monkeypatch):
...
     async with AsyncHeksherClient(fake_heksher_service.url, 10000000, ['a', 'b', 'c']) as client:
         response = await client.get_settings()
-        assert response == [
-            SettingData(name="settings",
-                        configurable_features=['a', 'b'],
-                        type='int',
-                        default_value=None,
-                        metadata={})
-        ]
+        assert response == {"settings": SettingData(name="settings",
```
Comment (bentheiii) [inline tests/unittest/test_async_client.py:281]: rename this example setting

File: heksher/clients/util.py [hunk 6]
```diff
@@ -82,32 +82,32 @@ def orjson_dumps(v, **kwargs):
...
-            settings_data.append(SettingData(
-                name=setting.name,
-                configurable_features=setting.configurable_features,
-                type=setting.type,
-                default_value=setting.default_value,
-                metadata=setting.metadata,
-                ))
-        return settings_data
+    def to_settings_data(self) -> Dict:
+        return {setting.to_setting_data().name: setting.to_setting_data() for setting in self.settings}
```
Comment (bentheiii) [inline heksher/clients/util.py:110]: avoid calling the same method twice

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,6 @@
 # heksher Changelog
 ## unreleased
+* Added get_settings method to thread and async clients
```
Comment (bentheiii) [inline CHANGELOG.md:3]:
```
### Added
```

**Review by bentheiii — APPROVED** [review]

**Pull request merged**